aten/src/ATen/cuda/CUDANumerics.cuh

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include <cuda.h>
+#include <limits.h>
+#include <assert.h>
+#include <typeinfo>
+
+// CUDANumerics.cuh contains mathematical functions that are either
+// not in the std namespace or are specialized for compilation
+// with CUDA NVCC or CUDA NVRTC. This header is derived from the
+// legacy THCNumerics.cuh.
+
+namespace at{
+
+template <typename T>
+struct numerics {
+};
+
+template <typename scalar_t>
+static inline __host__ __device__ scalar_t powi(scalar_t a, scalar_t b) {
+  // ge is only needed to not cause "pointless comparison warning"
+  assert(numerics<scalar_t>::ge(b, 0));
+  scalar_t result = 1;
+  while (b) {
+    if (b & 1) {
+       result *= a;
+    }
+    b /= 2;
+    a *= a;
+  }
+  return result;
+}
+
+template <>
+struct numerics<uint8_t> {
```

Comment:
I don't think we should have any of these functions. The std library functions should be sufficient.